lib/Controller/FolderApiV2Controller.php

```diff
@@ -0,0 +1,129 @@
+<?php
+/**
+ * Nextcloud - News
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ *
+ * @author    Paul Tirk <paultirk@paultirk.com>
+ * @copyright 2020 Paul Tirk
+ */
+
+namespace OCA\News\Controller;
+
+use \OCP\IRequest;
+use \OCP\IUserSession;
+use \OCP\AppFramework\Http;
+
+use \OCA\News\Service\FolderServiceV2;
+use \OCA\News\Service\ItemServiceV2;
+use \OCA\News\Service\Exceptions\ServiceNotFoundException;
+use \OCA\News\Service\Exceptions\ServiceConflictException;
+use \OCA\News\Service\Exceptions\ServiceValidationException;
+
+class FolderApiV2Controller extends ApiController
+{
+    use ApiV2ResponseTrait;
+
+    private $folderService;
+    private $itemService;
+
+    public function __construct(
+        IRequest $request,
+        IUserSession $userSession,
+        FolderServiceV2 $folderService,
+        ItemServiceV2 $itemService
+    ) {
+        parent::__construct($request, $userSession);
+
+        $this->folderService = $folderService;
+        $this->itemService = $itemService;
+    }
+
+    /**
+     * @NoAdminRequired
+     * @NoCSRFRequired
+     * @CORS
+     *
+     * @param string $name
+     * @return array|mixed|\OCP\AppFramework\Http\JSONResponse
+     */
+    public function createFolder($name)
+    {
+        try {
+            $this->folderService->purgeDeleted($this->getUserId(), false);
+            $responseData = $this->serialize(
+                $this->folderService->create($this->getUserId(), $name)
+            );
+            return $this->response([
+                'folder' => $responseData
+            ]);
+        } catch (ServiceValidationException $ex) {
+            return $this->errorResponse($ex, Http::STATUS_BAD_REQUEST);
+        } catch (ServiceConflictException $ex) {
```

Comment:
I'm not sure this can happen

Reply:
This is not finished, it was copied from the old API and I'm still not sure about those Exceptions, where they actually come frome, etc.

Reply:
Ok, about those exceptions. It is not thrown in the news app, if at all, then within nextcloud core. I could not trigger it though, only getting a status 500 instead of a 400. Should it be thrown here if the folder name is missing or is there another preferred way of validating the user input?

Reply:
@anoymouserver @SMillerDev, when you have time.. please..?

Reply:
Neither of these is actually going to happen in the service layer so no need to keep it.

Reply:
I see, but why are they used in e.g. `FolderServiceV2.php` then?`And the input checking should be done in the controller then?

Reply:
AFAIK they are not